app/controllers/posts_controller.rb

```diff
@@ -20,10 +20,10 @@ def create
     @post.likes_counter = 0
 
     if @post.save
-      flash[:success] = 'Post saved successfully'
+      flash[:alert] = 'Post saved successfully'
```

Comment:
Here in your `PostController` class inside the `index` method when you fetch the `post` at line 5 N+1 problem has to be solved. Consider trying something like this
```
@posts = @user.posts.includes(:comments)
```
That way all comments for that particular post will be fetched properly. Make sure after solving this problem your app is working as expected.